### PR TITLE
Minor refactoring in SqueezeNet

### PR DESCRIPTION
**Pull request by ekagra-ranjan**
1) Removed FIXME
~~2) Changed the `version` style parameter to `arch` parameter as in ResNets to remove the `arch = 'squeezenet' + version` statement.~~

### CODE REVIEW

File: torchvision/models/squeezenet.py [hunk 1]
```diff
@@ -36,10 +34,10 @@ def forward(self, x):
 
 class SqueezeNet(nn.Module):
 
-    def __init__(self, version='1_0', num_classes=1000):
+    def __init__(self, arch, num_classes=1000):
         super(SqueezeNet, self).__init__()
         self.num_classes = num_classes
-        if version == '1_0':
+        if arch == 'squeezenet1_0':
```
Comment (fmassa) [inline torchvision/models/squeezenet.py]: This is a BC-breaking change.

It's fine to do it for `ShuffleNetV2` because it hasn't been in any release yet, but SqueezeNet has been around for a while and this will break user code who relied on `SqueezeNet`.

File: torchvision/models/squeezenet.py [hunk 2]
```diff
@@ -36,10 +34,10 @@ def forward(self, x):
 
 class SqueezeNet(nn.Module):
 
-    def __init__(self, version='1_0', num_classes=1000):
+    def __init__(self, arch, num_classes=1000):
```
Comment (fmassa) [inline torchvision/models/squeezenet.py]: This is BC-breaking as well.

One way to address this would be to have both `version` and `arch`, and handle the case where `arch` is missing (old way).

Comment (ekagra-ranjan) [reply torchvision/models/squeezenet.py]: This would be a trouble only for users who are using `SqueezeNet()` function for their model and not the `squeezenet1_0` or `squeezenet1_1`, right?

Comment (fmassa) [reply torchvision/models/squeezenet.py]: Yes, that's correct. How many users call into the class instead of the modules? I don't know, but I have myself several times in the past inherited from models like `ResNet` to perform customization

Comment (ekagra-ranjan) [reply torchvision/models/squeezenet.py]: Using the class instead the modules for customization seems to be quite convenient. In that case I will revert the changes made to the `version` and let the removal of FIXME be only present. 

I think most of the users who use Pytorch as a framework read its docs and not the source code of the model. This hides the fact that some of the models support customized model implementation using the class itself. Do you think that adding a new module for accessing the class with customised hyperparameter along with adding them to the docs could be beneficial?

Comment (fmassa) [reply torchvision/models/squeezenet.py]: > Do you think that adding a new module for accessing the class with customised hyperparameter along with adding them to the docs could be beneficial?

In general, we pass `**kwargs` to the functions, so that the user only need to subclass the model class if we want to obtain something else out of it, e.g., returning intermediate activations of it.

I'm not clear if we should explicitly specify something else, could you clarify?

File: torchvision/models/squeezenet.py [hunk 3]
```diff
@@ -71,12 +69,6 @@ def __init__(self, version='1_0', num_classes=1000):
                 Fire(384, 64, 256, 256),
                 Fire(512, 64, 256, 256),
             )
-        else:
```
Comment (fmassa) [inline torchvision/models/squeezenet.py]: I think it is worthwhile to keep this.
For example, if the user passes `1.0` instead of `1_0`, this will silently dispatch to the `1_1` implementation.

File: torchvision/models/squeezenet.py (hunk 2 above)
Comment (ekagra-ranjan) [reply torchvision/models/squeezenet.py]: Suppose someone wants to make a custom resnet model with `planes=[3, 7, 14, 2]`. They can do that in the following way: 

`model = ResNet(inplanes=BasicBlock, planes=[3, 7, 14, 2], **kwargs)`

But the above functionality is not mentioned in the [docs](https://pytorch.org/docs/stable/torchvision/models.html). This customisation is not available in SqueezeNets but is available in DenseNets and ResNets. So would it do good if we could somehow mention this in the docs?

Comment (fmassa) [reply torchvision/models/squeezenet.py]: Sure, we could mention that in the docs.

But I'd assume that most users won't need this, because there will be no pre-trained models available for that particular configuration.
And for users who do want to try out a few different things, they will probably check the source code anyway, right? That's why having a simple to understand implementation is very important

Comment (ekagra-ranjan) [reply torchvision/models/squeezenet.py]: That does make sense! So should I close this PR?

Comment (fmassa) [reply torchvision/models/squeezenet.py]: I'm closing this PR then, but thanks anyway!

**Comment by fmassa** [comment]
Closing following discussion. Once again, thanks a lot @ekagra-ranjan for the PR!
